src/control-plane/backend/lambda/api/service/quicksight/sql-builder.ts

```diff
@@ -1853,36 +1843,36 @@ function _getStartDateForRelativeDateRange(
   const dayCount = Math.ceil(timeWindowInSeconds / 86400);
 
   if (timeUnit === ExploreRelativeTimeUnit.WK) {
-    return `DATEADD(DAY, -${dayCount}, date_trunc('week', current_date - interval '${lastN - 1} weeks'))` ;
+    return `DATEADD(DAY, -${dayCount}, date_trunc('week', CURRENT_TIMESTAMP AT TIME ZONE '${timezone}' - interval '${lastN - 1} weeks' ))`;
   } else if (timeUnit === ExploreRelativeTimeUnit.MM) {
-    return `DATEADD(DAY, -${dayCount}, date_trunc('month', current_date - interval '${lastN - 1} months'))`;
+    return `DATEADD(DAY, -${dayCount}, date_trunc('month', CURRENT_TIMESTAMP AT TIME ZONE '${timezone}' - interval '${lastN - 1} months' ))`;
   } else if (timeUnit === ExploreRelativeTimeUnit.YY) {
-    return `DATEADD(DAY, -${dayCount}, date_trunc('year', current_date - interval '${lastN - 1} years'))`;
+    return `DATEADD(DAY, -${dayCount}, date_trunc('year', CURRENT_TIMESTAMP AT TIME ZONE '${timezone}' - interval '${lastN - 1} years' ))`;
   } else {
-    return `DATEADD(DAY, -${dayCount}, date_trunc('day', current_date - interval '${lastN - 1} days'))`;
+    return `DATEADD(DAY, -${dayCount}, date_trunc('day', CURRENT_TIMESTAMP AT TIME ZONE '${timezone}' - interval '${lastN - 1} days' ))`;
   }
 }
 
 export function buildEventDateSql(sqlParameters: BaseSQLParameters, prefix: string = '', timeWindowInSeconds?: number) {
   let eventDateSQL = '';
   if (timeWindowInSeconds) {
     if (sqlParameters.timeScopeType === ExploreTimeScopeType.FIXED) {
-      eventDateSQL = eventDateSQL.concat(`CONVERT_TIMEZONE('${sqlParameters.timezone}', ${prefix}event_timestamp)::DATE >= date ${_getStartDateForFixDateRange(sqlParameters.timeStart, timeWindowInSeconds)} and CONVERT_TIMEZONE('${sqlParameters.timezone}', ${prefix}event_timestamp)::DATE <= date ${formatDateToYYYYMMDD(sqlParameters.timeEnd!)}`);
+      eventDateSQL = eventDateSQL.concat(`${prefix}event_timestamp >= (date ${formatDateToYYYYMMDD(sqlParameters.timeStart!)})::timestamp AT TIME ZONE '${sqlParameters.timezone}' - interval '${timeWindowInSeconds} seconds' and ${prefix}event_timestamp <= (date ${formatDateToYYYYMMDD(sqlParameters.timeEnd!)} + interval '1 days' )::timestamp AT TIME ZONE '${sqlParameters.timezone}'`);
```

Comment:
Why add 1 day in `end_time`?

Reply:
add 1 day is ok due to we just use yyyymmdd part of the time. if not , we need append ‘23:59:59’ to the timestamp